src/main/java/com/google/roomies/data/Comment.java

```diff
@@ -0,0 +1,67 @@
+package com.google.roomies;
+ 
+import static com.google.roomies.CommentRequestParameterNames.COMMENT;
+import static com.google.roomies.CommentRequestParameterNames.TIMESTAMP;
+
+import com.google.api.core.ApiFuture;
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Preconditions;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.FieldValue;
+import com.google.cloud.Timestamp;
+import com.google.common.collect.ImmutableMap;
+import com.google.roomies.database.DatabaseFactory;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import javax.servlet.http.HttpServletRequest;
+
+/** A comment made by a user under a listing. */
+@AutoValue
+public abstract class Comment {
+  abstract Optional<String> documentId();
+  abstract Optional<Timestamp> timestamp();
+  public abstract String comment();
```

Comment:
Changed to commentMessage. commentMessage() is public so I can get the message for the error message in the database class if the listing doesn't exist since the database classes are in a separate package